### ISSUE TITLE
[Feature] Support for keyboard arrow keys

### ISSUE THREAD

**Issue by selvam920** [maheshmnj/searchfield#7]
Pls support arrow keys down and up and enter key

**Comment by maheshmnj** [comment]
H @selvam920, could you please provide some more details about the expected behavior.

Thnaks

**Comment by wishstore36** [comment]
> H @selvam920, could you please provide some more details about the expected behavior.
> 
> Thanks

In case users presses down / up arrow key , the selection within the options must be selected appropriately and on press of "Enter" key, the option selected must be returned, apart from down / up arrow keys, it will be gr8 to support page up and page down to scroll within the suggestion list.

**Comment by maheshmnj** [comment]
@wishstore36, thanks for the clarification that makes sense to me and it would be a great feature.

**Comment by lucasjinreal** [comment]
@maheshmnj  hi, if so, than have listen to the keyboard press event could be cubersome.

**Comment by JayGoti** [comment]
Is this feature added in latest version?

**Comment by maheshmnj** [comment]
Quick update on this issue, we can now navigate list items using shift + tab and up/down arrow keys and select a list item using return

https://github.com/maheshmnj/searchfield/assets/31410839/f7d1e75b-1dad-454d-8b8a-adffab63bb3d

This needs few additional improvements

- [ ] When the list is in focus we cannot search
- [ ] navigating the list item should keep the focused item in the view (list should scroll as we navigate)

You should be able to test this by switching to `issue-7` branch by adding the below dependency to your pubspec.yaml

```yaml
searchfield:
    git:
      url: git://github.com/maheshmnj/searchfield.git
      ref: issue-7
```

**Comment by lucasjinreal** [comment]
thanks, would releases on pub.dev?

**Comment by maheshmnj** [comment]
Soon once the two issues listed in the above comment are resolved, meanwhile you should be able to use it in your app by specifying the git URL mentioned in the above comment.

**Comment by maheshmnj** [comment]
The current implementation is incorrect, At any instance, only one widget can remain in focus as mentioned here by one of the Flutter engineers https://discord.com/channels/608014603317936148/608014603317936150/1214604507015553086

We should probably keep the textfield in focus all the time and move the highlight based on the current selectedIndex.

**Comment by maheshmnj** [comment]
Folks this is now released in [v0.9.5](https://pub.dev/packages/searchfield), try it out!
